frontend/src/components/TraineeForm/TraineeForm.jsx

```diff
@@ -0,0 +1,179 @@
+import React, { useState } from "react";
+import "./TraineeForm.css";
+import Button from "../shared/Button";
+
+const formInitialState = {
+  trainee_name: "",
+  github_link: "",
+  portfolio_link: "",
+  linkedIn_link: "",
+  role: "",
+  about_me: "",
+  skills: "",
+};
+
+const TraineeForm = () => {
+  const [formDetails, setFormDetails] = useState(formInitialState);
+
+  function handleInputChange(event) {
+    event.preventDefault();
+    setFormDetails({ ...formDetails, [event.target.name]: event.target.value });
+  }
+
+  async function postFormData(data) {
+    try {
+      const response = await fetch(
+        "http://127.0.0.1:5000/submit_trainee_form",
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(data),
+        }
+      );
+      if (!response.ok) {
+        throw new Error(await response.json());
+      }
+      const result = await response.json();
+      console.log("Success:", result);
+      setFormDetails(formInitialState);
+      return result;
+    } catch (error) {
+      console.log("Something went wrong:", error);
+      throw error;
+    }
+  }
+
+  async function handleSubmit(event) {
+    event.preventDefault();
+
+    const data = {
+      trainee_name: formDetails.trainee_name,
+      github_link: formDetails.github_link,
+      portfolio_link: formDetails.portfolio_link,
+      linkedIn_link: formDetails.linkedIn_link,
+      role: formDetails.role,
+      about_me: formDetails.about_me,
+      skills: formDetails.skills,
+    };
+    try {
+      await postFormData(data);
```

Comment:
I dont think there is any need to create a new object here
```suggestion
    try {
      await postFormData(formDetails);
```